lib/web_apis/web_apis.es6.js

```diff
@@ -0,0 +1,253 @@
+            {interfaceName: {apiName: [Boolean ...]} ...}
+            It contains information about browsers and their interface/API.`,
+      },
+      code: function(browserAPIs, browserKeys) {
+        let apiMap = {};
+        let browserDict = {}; // O(1) lookup for browser's index in header.
+        apiMap._header = browserKeys.slice();
+        let numBrowsers = apiMap._header.length;
+        // Fill in browserDict.
+        for (let i = 0; i < numBrowsers; i++) {
+          browserDict[browserKeys[i]] = i;
+        }
+        // Fill in apiMap.
+        for (let i = 0; i < browserAPIs.length; i++) {
+          let interfaceName = browserAPIs[i].interfaceName;
+          let apiName = browserAPIs[i].apiName;
+          if (!apiMap.hasOwnProperty(interfaceName)) {
+            // If typeof is not object, this may be a build-in function,
+            // overwrite it.
+            apiMap[interfaceName] = {};
+          }
+          if (!apiMap[interfaceName].hasOwnProperty(apiName)) {
+            // If visit apiName for first time,
+            // initialize an array of false.
+            apiMap[interfaceName][apiName] = [];
+            for (let k = 0; k < numBrowsers; k++) {
+              apiMap[interfaceName][apiName].push(false);
+            }
+          }
+          apiMap[interfaceName][apiName][browserDict[
+            browserAPIs[i].browserKey]] = true;
+        }
+        return(apiMap);
+      },
+    },
+    {
+      name: 'toMap',
+      documentation: `An asynchronous method that takes an array of browser
+        keys, produce a JSON containing browser and interface relationship.
+        This structure should be easy to be displayed as a nested table in
+        client side.`,
+      args: [
+        {
+          name: 'browserKeys',
+          typeName: 'StringArray',
+          documentation: `A string array contains valid browser keys`,
+        },
+      ],
+      returns: {
+        typeName: 'JSON',
+        documentation: `Returns a JSON of the form:
+                {interfaceName: {apiName: [Boolean ...]} ...}
+                This structure is easy for displaying a nested table.`,
+      },
+      code: function(browserKeys) {
+        return this.browserAPIs
+          .where(this.mlang.IN(this.BrowserAPI.BROWSER_KEY, browserKeys))
+          .select()
+          .then((arraySink) => {
+            return(this.toMap_(arraySink.a, browserKeys));
+          });
+      },
+    },
+    {
+      name: 'toCSV',
+      documentation: `Takes an array of browser keys, produce a string of
+        CSV format.`,
+      args: [
+        {
+          name: 'browserKeys',
+          typeName: 'StringArray',
+          documentation: `A string array contains valid browser keys.`,
+        },
+      ],
+      returns: {
+        typeName: 'String',
+        documentation: `Returns a string of csv format.`,
+      },
+      code: function(browserKeys) {
+        return this.toMap(browserKeys).then(function(result) {
```

Comment:
Can't we just:

```
this.browserAPIs.where(this.IN(this.BrowserAPI.BROWSER_KEY, browserKeys))
  .select(this.GROUP_BY(this.BrowserAPI.INTERFACE_KEY))
  .then(function() { /* Spit out rows according to missing/not-missing objects in groups */ })
```

?

Of course, we'll need to output the header first, but that's fine.